avocado/core/job.py

```diff
@@ -103,13 +98,9 @@ def resolutions_to_tasks(resolutions, config):
                         config.get("{}.filter_by_tags_include_empty".format(sub_cmd)),
                         config.get('{}.filter_by_tags_include_empty_key'.format(sub_cmd))):
                     continue
-            if runnable.uri:
-                name = runnable.uri
-            identifier = str(TestID(index + 1, name, None, no_digits))
             status_server = config.get('nrun.status_server.listen')
-            tasks.append(nrunner.Task(identifier, runnable,
+            tasks.append(nrunner.Task(None, runnable,
```

Comment:
If `identifier` is optional, we could consider change this class signature and add the optional arguments to the end. As far as I could count, there are less than 10 calls to `Task()` on the current code,  but not sure how difficult it is to change this now.

Reply:
> If `identifier` is optional, we could consider change this class signature and add the optional arguments to the end. As far as I could count, there are less than 10 calls to `Task()` on the current code, but not sure how difficult it is to change this now.

@beraldoleal what if we use a `uuid` of sorts to ensure that each task will be "ready", even if outside of the Job concept?  We could either do that at Task initialization (`if identifier is None: identifier = uuid()`, or for a one-off and simpler change, switch `None` here for a `uuid`.  The reason is that I still feel like a task *should always have* a unique identifier... its value is pretty limited without one.

Reply:
I'm ok with the second approach: `Task(uuid(), ...)` so we can avoid the first argument being optional. And if we decide in the future that the id is not required we can change the signature.